### PR TITLE
use dotnet 8 preview sdk and lang preview

### PR DESCRIPTION
**Pull request by dawedawe**
To increase our level of dogfooding
- use dotnet 8 preview sdk
- use lang preview

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -34,6 +34,7 @@ Some common use cases include:
         <!-- https://www.gresearch.co.uk/blog/article/improve-nuget-restores-with-static-graph-evaluation/ -->
         <RestoreUseStaticGraphEvaluation>true</RestoreUseStaticGraphEvaluation>
         <ServerGarbageCollection>true</ServerGarbageCollection>
+        <LangVersion>PREVIEW</LangVersion>
```
Comment (nojaf) [inline Directory.Build.props:37]: Is it `PREVIEW` in all caps? Or does that not matter at all?

Comment (auduchinok) [reply Directory.Build.props]: It doesn't matter. Normally, tooling inserts lowercase.

Comment (dawedawe) [reply Directory.Build.props]: Yes, changed it to lowercase to be more consistent.

**Comment by nojaf** [comment]
The documentation fails, I guess `fsdocs` would need FCS `43.8.100-preview.23327.15` perhaps.
I'll take a look at what needs to be done. Maybe it is a bump in proj-info to deal with dotnet 8 sdk.

**Comment by dawedawe** [comment]
> The documentation fails, I guess `fsdocs` would need FCS `43.8.100-preview.23327.15` perhaps. I'll take a look at what needs to be done. Maybe it is a bump in proj-info to deal with dotnet 8 sdk.

Thanks!

**Comment by nojaf** [comment]
There still is something fishy with the project cracking happening during CI.
We would need to tackle this in `Proj-Info` some day.

**Comment by dawedawe** [comment]
> There still is something fishy with the project cracking happening during CI. We would need to tackle this in `Proj-Info` some day.

Can you reproduce that locally on your machine?
Asking because I can't. Maybe it's an issue with the build environment...

**Comment by nojaf** [comment]
I'm bumping into https://github.com/ionide/proj-info/issues/195 I think.

**Review by nojaf — APPROVED** [review]
This is good to go now!
Turns out the docs need some magic envs to use the correct `System.Runtime` jazz.

**Pull request merged**